merkletree/src/merkle/proof/multi/mod.rs

```diff
@@ -292,7 +307,7 @@ impl MultiProofHashes {
         // Note: This can be made more efficient by marking "hashed" nodes and skipping them,
         // but we don't care about performance here. We care more about security.
         for (leaf_index, leaf_hash) in &leaves {
-            let mut hash = *leaf_hash;
+            let mut hash = leaf_hash.clone();
```

Comment:
It may be possible to replace some of these `.clone()`s with moves

Reply:
I gave it a shot yesterday. Didn't work out. The only place where this can be done is at the end of the loop because the value is left just to do the `==`... but the problem there is I have to mess around and do the equality before the clone, which is not clean and I'm afraid it could cause other issues in the future. I'm hoping the compiler could be smart about this.